lib/OAuth2.php

```diff
@@ -888,12 +888,8 @@ protected function grantAccessTokenAuthCode(IOAuth2Client $client, array $input)
             throw new OAuth2ServerException(self::HTTP_BAD_REQUEST, self::ERROR_INVALID_GRANT, "Code doesn't exist or is invalid for the client");
         }
 
-        // Validate the redirect URI. If a redirect URI has been provided on input, it must be validated
-        if ($input["redirect_uri"] && !$this->validateRedirectUri(
-                $input["redirect_uri"],
-                $authCode->getRedirectUri()
-            )
-        ) {
+        // Validate the redirect URI. If a redirect URI has been provided on input, it must be identical
+        if ($input["redirect_uri"] && $input["redirect_uri"] === $authCode->getRedirectUri()) {
```

Comment:
Shouldn't this be `!==` for "fail if `redirect_uri` is set and is **_not_** equal to the stored one"?